### DISCUSSION TITLE
Algorithms and data structures

### DISCUSSION THREAD

**Discussion by hatemhosny** [gleam-lang/gleam#3084] [Ideas & suggestions]
Consider providing examples for common algorithms and data structures. 

I suggest adding that as a repo in [TheAlgorithms](https://github.com/TheAlgorithms). These are some potential benefits:
- Developers can see how Gleam solves common problems as compared to other languages. 
- The organization is very popular (374k total stars). This can increase your visibility.
- They have a [website](https://the-algorithms.com/) that shows the implementations and allows running the code on an online playground ([example](https://the-algorithms.com/playground?algorithm=binary-search&language=javascript)). They use the [LiveCodes](https://github.com/live-codes/livecodes) playground, which already [supports Gleam](https://github.com/gleam-lang/gleam/discussions/3041). You can use that as reference for demos and examples.

**Comment by lpil** [comment]
This would be a great project for you if you are interested in it! The core team is busy working on Gleam.
